### ISSUE TITLE
`0 0 1-3,12 * * *` good, `0 0 12,1-3 * * *` bad

### ISSUE THREAD

**Issue by KevinWang15** [harrisiirak/cron-parser#179]
I think there is a bug here, both `0 1-3,12 * * *` and `0 12,1-3 * * *` should produce 
```
Fri Jan 17 2020 01:00:00 GMT+0000
Fri Jan 17 2020 02:00:00 GMT+0000
Fri Jan 17 2020 03:00:00 GMT+0000
Fri Jan 17 2020 12:00:00 GMT+0000
Sat Jan 18 2020 01:00:00 GMT+0000
Sat Jan 18 2020 02:00:00 GMT+0000
Sat Jan 18 2020 03:00:00 GMT+0000
```

however, `0 12,1-3 * * *` produces
```
Fri Jan 17 2020 12:00:00 GMT+0000
Sat Jan 18 2020 12:00:00 GMT+0000
Sun Jan 19 2020 12:00:00 GMT+0000
Mon Jan 20 2020 12:00:00 GMT+0000
Tue Jan 21 2020 12:00:00 GMT+0000
Wed Jan 22 2020 12:00:00 GMT+0000
Thu Jan 23 2020 12:00:00 GMT+0000
```

**Comment by KevinWang15** [comment]
my quick & dirty fix:

```patch
Index: lib/expression.js
<+>UTF-8
===================================================================
--- lib/expression.js	(revision 6eae107aef5cb137858ae5a79e8385c36e257b6b)
+++ lib/expression.js	(date 1579182013534)
@@ -251,7 +251,16 @@
       }
     }
 
-    var atoms = val.split(',');
+    var atoms = val.split(',').sort((a, b) => {
+      const getFirstIfIsArray = (val) => {
+        if (Array.isArray(val)) {
+          return val[0];
+        }
+        return val;
+      };
+      return getFirstIfIsArray(parseRepeat(a)) - getFirstIfIsArray(parseRepeat(b));
+    });
+
     if (atoms.length > 1) {
       for (var i = 0, c = atoms.length; i < c; i++) {
         handleResult(parseRepeat(atoms[i]));
```

**Comment by harrisiirak** [comment]
@KevinWang15 thanks for reporting this! Yeah, this looks like a bug. I'll look into this coming days.

**Comment by harrisiirak** [comment]
@KevinWang15 sorry about the delay. Pushed the fix in 110a823. I'm using `localeCompare` to sort atoms before processing.
